### ISSUE TITLE
Wrong tag in simplebuffer?

### ISSUE THREAD

**Issue by timfel** [AmigaPorts/ACE#170]
Simplebuffer has the same VPM ID as scrollbuffer. That means I cannot look one up. Is that intended?

https://github.com/AmigaPorts/ACE/blob/7f82b9995ec0aa306a67744bf7b0c9fb1bfa88de/src/ace/managers/viewport/simplebuffer.c#L154

**Comment by tehKaiN** [comment]
I kinda wrongly named that - it should be a category instead of ID. My idea was that each vport manager falls into specific category which enforces order of processing and you shouldn't have two managers of same category on single vport.

Do you need to query in runtime whether you have simplebuffer or scrollbuffer?

**Comment by timfel** [comment]
I'm building a new buffer manager based around simplebuffer instead of a scrollbuffer, and I was taking inspiration from the tilebuffer and how it wraps a scrollbuffer.

**Comment by tehKaiN** [comment]
yeah, you should construct a simplebuffer with your manager, I guess - same as tilebuffer constructs scrollbuffer. What's your plan / intended principle of operation? perhaps I can suggest something better.

**Comment by timfel** [comment]
The plan is to have a tilemap display driven entirely by the copper. A bit like a char display on a c64, where I would just modify one byte, and the entire bitmap for that manager is redrawn on the next frame without CPU, driven by the copper.

**Comment by tehKaiN** [comment]
Whoa, good luck with that! I don't think it'll work though - copper will eat some chip bus bandwidth and I don't think you can redraw the whole screen each frame, especially when doing small tile-sized blits - unless you target 25fps or something. Unless you want to do some selective redraws, but I guess that's gonna take more than one change on the copperlist data. Looking forward to your results, I'm a bit puzzled, but very interested. :)

I guess you should use your own VPM which is higher than simple buffer's VPM in order for your things to work after processing of simplebuffer. If you want to do it otherwise, you can either multiply ACE's VPM categories by 10 or something so that there are spaces between them and inject your VPM with lower category than simplebuffer's. If that's the case, I'll of course accept the PR with changed VPM values.

**Comment by timfel** [comment]
Well, I've been playing with it a bit, and on an empty map the tilebuffer is already too slow for my taste even when I'm just using the arrow keys ;) I had to slow the camera down to 4px per frame to get diagonal redraw to be fast enough, and that feels very slow to me :)
